wandb/apis/public.py

```diff
@@ -287,15 +287,22 @@ class Api(object):
     )
 
     def __init__(self, overrides={}, timeout: Optional[int] = None):
-        self.settings = InternalApi().settings()
-        if self.api_key is None:
-            wandb.login()
-        self.settings.update(overrides)
+
         if "username" in overrides and "entity" not in overrides:
             wandb.termwarn(
                 'Passing "username" to Api is deprecated. please use "entity" instead.'
             )
-            self.settings["entity"] = overrides["username"]
+            overrides["entity"] = overrides["username"]
+
+        _api = InternalApi()
+        for key, value in overrides.items():
+            _api.set_setting(key, value)
+
+        self.settings = _api.settings()
+
+        if self.api_key is None:
```

Comment:
@vanpelt @raubitsj I think this is the correct order. `api_key` depends on `self.settings['base_url']` so in the old way we would check for the old settings and then update it. Seems wrong to me, I assume the `base_url` should be the same for login and for client